pkg/webhook/server/certificate/manager_test.go

```diff
@@ -1,184 +1,185 @@
-	}
+			service = &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "fooWebhook",
+					Name:      "fooWebhook",
+				},
+			}
+		)
+		BeforeEach(func() {
+			By("Init fake client with mutatingwebhookconfiguration and service")
+			objs := []runtime.Object{mutatingWebhookConfiguration, service}
+			client = fake.NewFakeClient(objs...)
+
+			By("Calling waitForDeadlineAndRotate for the first time")
+			manager = NewManager(client, "fooWebhook", MutatingWebhook, certsDuration)
+			manager.waitForDeadlineAndRotate()
+			//TODO Implement ErrorsHandler to take the errors that we have at
+			//     background
 
-	secret := corev1.Secret{}
-	err = client.Get(context.TODO(), types.NamespacedName{Name: "fooWebhook", Namespace: "fooWebhook"}, &secret)
-	if err != nil {
-		t.Fatalf("get secret: (%v)", err)
-	}
+		})
+		It("should create the secret and rotate CA and server certificates", func() {
 
-	if secret.Type != corev1.SecretTypeTLS {
-		t.Fatalf("Non TLS secret type %s", secret.Type)
-	}
+			err := client.Get(context.TODO(), types.NamespacedName{Name: "fooWebhook"}, mutatingWebhookConfiguration)
+			Expect(err).ToNot(HaveOccurred(), "should success getting mutatingwebhookconfiguration")
 
-	if len(secret.Data) == 0 {
-		t.Fatal("No tls key/cert at secret")
-	}
+			clientConfig := mutatingWebhookConfiguration.Webhooks[0].ClientConfig
+			Expect(clientConfig.CABundle).ToNot(BeEmpty(), "should update CA budle")
 
-	nextRotation := time.Now().Sub(manager.nextRotationDeadline())
-	start := time.Now()
-	err = wait.PollImmediate(nextRotation, nextRotation, func() (bool, error) {
-		manager.waitForDeadlineAndRotate()
-		return true, nil
+			secret := corev1.Secret{}
+			err = client.Get(context.TODO(), types.NamespacedName{Name: "fooWebhook", Namespace: "fooWebhook"}, &secret)
+			Expect(err).ToNot(HaveOccurred(), "should success getting secret")
+			Expect(secret.Type).To(Equal(corev1.SecretTypeTLS), "should be a TLS secret")
+			Expect(secret.Data).ToNot(BeEmpty(), "should contain a secret with TLS key/cert")
+		})
+		Context("and called a second time", func() {
+			var (
+				secret       corev1.Secret
+				start        time.Time
+				clientConfig admissionregistrationv1beta1.WebhookClientConfig
+			)
+			BeforeEach(func() {
+				err := client.Get(context.TODO(), types.NamespacedName{Name: "fooWebhook"}, mutatingWebhookConfiguration)
+				Expect(err).ToNot(HaveOccurred(), "should success getting mutatingwebhookconfiguration")
+				clientConfig = mutatingWebhookConfiguration.Webhooks[0].ClientConfig
+
+				err = client.Get(context.TODO(), types.NamespacedName{Name: "fooWebhook", Namespace: "fooWebhook"}, &secret)
+				Expect(err).ToNot(HaveOccurred(), "should success getting secret")
+
+				start = time.Now()
+				By("Calling waitForDeadlineAndRotate for the second time")
+				manager.waitForDeadlineAndRotate()
+			})
+
+			It("should wait some time, update the secret and re-rotate certificates", func() {
```

Comment:
I don't understand something here.
we expect that certs should be rotated, but it is not clear why they've been rotated (expired?/secret deleted?/else?)
it should be clear from the Context what type of re-rotation we are testing.